packages/@react-aria/color/src/useColorArea.ts

```diff
@@ -0,0 +1,355 @@
+        if (e.pointerType === 'mouse' && (e.button !== 0 || e.altKey || e.ctrlKey || e.metaKey)) {
+          return;
+        }
+        onColorAreaDown(e.currentTarget, e.pointerId, e.clientX, e.clientY);
+      }} : {
+        onMouseDown: (e: React.MouseEvent) => {
+          if (e.button !== 0 || e.altKey || e.ctrlKey || e.metaKey) {
+            return;
+          }
+          onColorAreaDown(e.currentTarget, undefined, e.clientX, e.clientY);
+        },
+        onTouchStart: (e: React.TouchEvent) => {
+          onColorAreaDown(e.currentTarget, e.changedTouches[0].identifier, e.changedTouches[0].clientX, e.changedTouches[0].clientY);
+        }
+      })
+  }, movePropsContainer);
+
+  let thumbInteractions = isDisabled ? {} : mergeProps({
+    ...(typeof PointerEvent !== 'undefined' ? {
+      onPointerDown: (e: React.PointerEvent) => {
+        if (e.pointerType === 'mouse' && (e.button !== 0 || e.altKey || e.ctrlKey || e.metaKey)) {
+          return;
+        }
+        onThumbDown(e.pointerId);
+      }} : {
+        onMouseDown: (e: React.MouseEvent) => {
+          if (e.button !== 0 || e.altKey || e.ctrlKey || e.metaKey) {
+            return;
+          }
+          onThumbDown(undefined);
+        },
+        onTouchStart: (e: React.TouchEvent) => {
+          onThumbDown(e.changedTouches[0].identifier);
+        }
+      })
+  }, keyboardProps, movePropsThumb);
+
+
+  let xInputLabellingProps = useLabels({
+    ...props,
+    'aria-label': `${state.value.getChannelName(xChannel, locale)} / ${state.value.getChannelName(yChannel, locale)}`
+  });
+
+  let yInputLabellingProps = useLabels({
+    ...props,
+    'aria-label': `${state.value.getChannelName(xChannel, locale)} / ${state.value.getChannelName(yChannel, locale)}`
+  });
+
+  let colorAriaLabellingProps = useLabels(props);
+
+  let getValueTitle = () =>  [
+    `${state.value.getChannelName('red', locale)}: ${state.value.formatChannelValue('red', locale)}`,
+    `${state.value.getChannelName('green', locale)}: ${state.value.formatChannelValue('green', locale)}`,
+    `${state.value.getChannelName('blue', locale)}: ${state.value.formatChannelValue('blue', locale)}`
+  ].join(', ');
+
+  let ariaRoleDescription = messages.getStringForLocale('twoDimensionalSlider', locale);
+
+  let {visuallyHiddenProps} = useVisuallyHidden({style: {
+    opacity: '0.0001',
+    width: '100%',
+    height: '100%',
+    pointerEvents: 'none'
+  }});
+
+  return {
+    colorAreaProps: {
+      ...colorAriaLabellingProps,
+      ...colorAreaInteractions,
+      role: 'group'
+    },
+    gradientProps: {},
+    thumbProps: {
+      ...thumbInteractions
+    },
+    xInputProps: {
+      ...xInputLabellingProps,
+      ...visuallyHiddenProps,
+      type: 'range',
+      min: state.value.getChannelRange(xChannel).minValue,
+      max: state.value.getChannelRange(xChannel).maxValue,
+      step: xChannelStep,
+      'aria-roledescription': ariaRoleDescription,
```

Comment:
For VoiceOver on iOS or Talkback on Android, I'm starting to question whether we need or should add `aria-roledescription="2D slider"` to the `input[type="range"]`.  For the desktop use  case, the role description makes sense, but with a mobile screen reader, each input receives focus independently and the swipe gesture to increment or decrement the value of the focused input is one-dimensional, so calling it a "2D slider" may create confusion. The only use case where "2D slider" makes sense with a mobile screen reader is double-tap and hold to create a pass-through gesture, which is a rarely used interaction pattern.